### ISSUE TITLE
Setup code.quarkus.io template URLs

### ISSUE THREAD

**Issue by maybeec** [devonfw-forge/devonfw-microservices#11]
As a user I would like to get to know how to start with quarkus for my microservice in order to simply start by with the right standards and technologies.

Quarkus Code Generator is providing a lot of alternatives on technologies and libraries to be integrated.
We should guide the developer/architect to a basic set of frameworks to be integrated when he/she is starting a project. For example which REST API technology to use? - by simply providing pre-selected URL to the generator, the user can anyhow change the selection after going to the quarkus generator page. But he already got the recommendations from devonfw pre-selected.

**Comment by yntelectual** [comment]
In the long run, maybe it would make sense to create a custom archetype in Devon - e.g. archetype for backend microservices with curated deps(ORM, JAXRS, REST client, DB changelog, JWT, Validation,  Health, Metrics, Telemetry, custom extensions... ) and sample code, infra setup(local dev + docker, etc) which would make it more stable and ready to start for devs.

**Comment by SchneiderMalte** [comment]
RESTEasy JAX-RS: Standard in creation of projects, also reactive
RESTEasy Jackson: For REST services to consume and produce JSON payloads, also reactive
RESTEasy JSON-B: Alternative to Jackson. Also reactive. Jackson state-of-the-art https://itsallbinary.com/java-json-polymorphic-type-handling-jackson-vs-gson-vs-json-b/
RESTEasy JAXB: Transformation of responses to XML
RESTEasy Multipart: multipart mime format is used to pass lists of content bodies
Eclipse Vert.x GraphQL: API Querying, most functions covered by REST https://blog.logrocket.com/why-you-shouldnt-use-graphql/
SmallRye GraphQL: MicroProfile GraphQL
Hibernate Validator: Bean validation 
JAX-RS Client Reactive: Reactive
Rest clients: Postman already for devonfw (but not listed here)
SmallRye OpenAPI: API doc with Swagger
gRPC: Alternative to REST. https://cloud.google.com/blog/products/api-management/understanding-grpc-openapi-and-rest-and-when-to-use-them
REST resources for Hibernate ORM with Panache: Cleaner code ext. for Hibernate, (also possible for Mongodb)
RESTEasy SmallRye Mutiny: reactive programming library allowing to express and compose asynchronous actions
Mutiny support for REST Client: async event messaging
RESTEasy Qute: Reactive HTML templates
SmallRye Reactive Messaging: Kafka integration
Reactive Routes: 
SmallRye JWT: JWT management
Undertow Servlet: Web server alternative to Tomcat. Performs better? https://examples.javacodegeeks.com/enterprise-java/spring/tomcat-vs-jetty-vs-undertow-comparison-of-spring-boot-embedded-servlet-containers/
WebJar Locator: Client dependency packaging in JAR files. Alternative to yarn? https://dzone.com/articles/why-using-webjars-is-a-good-idea
JDBC Drivers for different db
Agroal: Datasource and connection pooling (already in Hibernate)
AWS extensions

First understanding: Different extensions for different use cases. Only some compete with each other.

**Comment by SchneiderMalte** [comment]
mvn io.quarkus:quarkus-maven-plugin:1.13.2.Final:create “-DprojectGroupId=org.acme” -DprojectArtifactId=devon-quarkus-starter -DclassName="org.acme.quarkus.starter.GreetingResource" -Dextensions="resteasy,resteasy-jackson,hibernate-orm,jdbc-postgresql,undertow,hibernate-validator,vertx,kubernetes,jib,smallrye-health,smallrye-fault-tolerance,minikube,micrometer-registry-prometheus,oidc"

**Comment by SchneiderMalte** [comment]
topic | detail | Standard   (non-reactive) | alternative   implementation | comment
-- | -- | -- | -- | --
runtime | servlet-container | Undertow | use in camel context |  
component management | dependency   injection | JSR 365, ArC (provides interceptors   that can be used to implement the same functionalities as AOP provides) |   |  
configuration |   | SmallRye config, Eclipse MicroProfile   Config (also supports YAML) |   |  
persistence | OR-mapper | JPA, hibernate |   |  
storage | database | JDBC driver for Postgres as standard? | Multiple (un-) structured db |  
batch | - | - |   | JSR 352 planned
service | SOAP services |   |   | In camel context
service | REST services | RESTEasy   JAX-RS JSR-311, RESTEasy Jackson for JSONs   Clients needed? | RESTEasy JSON-B for JSONs. RESTEasy JAXB   for XML. RESTEasy Multipart for content bodies   gRPC | Most available as reactive and in camel.   Documentable via swagger openAPI
service | Async   messaging | SmallRye Reactive   Messaging, Vert.x EventBus |   | Partly reactive but essential part of MS.
Cloud | Kubernetes | Kubernetes | OpenShift |  
Cloud | Deployment | Minikube |   | Worked fine for me without the extension   (but assumingly makes deploying easier)
logging | framework | JBoss Log Manager and the JBoss Logging facade | use in camel context | They merge logs from: JUL,   JBoss   Logging, SLF4J, Apache Commons   Logging
validation | framework | beanvalidation/JSR380,   hibernate validator impl. | use in camel context | Also, validation of json and xml payloads   with camel
security | Authentication   & Authorization | OpenID Connect   for services and web applications. | Smallrye-jwt for providing   json web tokens   OAuth2 Authentication | Comparison
monitoring | framework | Micrometer metrics | SmallRye Metrics: when   it’s required to retain MicroProfile specification compatibility   use in camel context |  
monitoring | Health | SmallRye Health, impl of   MicroProfile health | use in camel context |  
monitoring | Fault Tolerance | SmallRye Fault   Tolerance: implementation of the MicroProfile Fault Tolerance | use in camel context |

**Comment by SchneiderMalte** [comment]
Basic URL: https://code.quarkus.io/?g=org.devonfw&e=resteasy&e=resteasy-jackson&e=hibernate-validator&e=hibernate-orm&e=micrometer

**Comment by hohwille** [comment]
What really mattest from a "devon4j perspective" is that we try to keep the same APIs and standards:
* JAX-RS for REST services
* SLF4J as logging API
* JSR-330, JSR-250, optionally JSR-365 for dependency-injection and container bean management
* Java Persistence API (JPA)
* Spring-Data-JPA (see #37)
* JSR-250 for functional authorization (`@RolesAllowed`)
* JSR-303 (Bean Validation) for syntactic validation
* servlet standard
* etc.

All this seems to be aligned.
The implementations as secondary. In the end they surely somehow matter for non-functional aspects but with quarkus the choice of technology is not as flexible as with spring so we have to follow the framework.
That means as far as I understood:
* Apache CXF is not supported by quarkus so we have to choose RESTEasy
* sping-* (except for spring-data) will not be used with quarkus but instead we use the quarkus specific solutions (so e.g. no spring-security).

To summarize all looks sane to me.
The only option I am missing is `spring-data-jpa` but it is an option/alternative to plain DAOs with JPA/hibernate.
However, I expect this to become the recommended default in "devon4quarkus".

**Comment by hohwille** [comment]
> Authentication | OpenID Connect   for services and web applications. | Smallrye-jwt for providing   json web tokens   OAuth2 Authentication

My understanding is that we as a foundation and standard only rely on JWT authentication.
So our app (microservice) is not by default required to support OpenID Connect or OAuth2 what are both rather complex standards with many features.
